Dynatrace/dynatrace-operator
# Simplify make deploy/undeploy commands

# Description

Removes code duplication for the `make deploy/x` and `make undeploy/x` commands.
Unfortunately, the same can't really be done for `make manifests/x` as the commands are different just enough to make generalising it more annoying then its worth. 

## How can this be tested?
Run effected commands:
`make deploy/kubernetes`
`make deploy/kubernetes/no-csi`
`make deploy/openshift`
`make deploy/openshift/no-csi`
`make undeploy/kubernetes`
`make undeploy/kubernetes/no-csi`
`make undeploy/openshift`
`make undeploy/openshift/no-csi`
`make deploy/gke-autopilot`
`make undeploy/gke-autopilot`

Run unchanged commands, to make sure we didn't mess them up:
`make deploy/helm`
`make undeploy/helm`


## Checklist
